src/components/shared/Terminal.tsx

```diff
@@ -104,15 +113,131 @@ function Terminal(prop: {
+      if (path === '/') {
+        // TODO: Figure out how to handle this
+        return ["What are you doing? You can't delete the root directory!"];
+      }
+      dir.removeFileSystemObject(fileName);
+    } else {
+      if (directory && !file.isEmpty()) {
+        return [`rmdir: '${path}': Directory not empty`];
+      }
+      return [`rm: '${path}': is a directory`];
+    }
+
+    prop.setFileSystem(fileSystemCopy);
+    prop.setCurrentWorkingDirectory(cwdCopy);
+    return [];
+  }
+
+  function executeRmdir(path: string, flags: string): string[] {
+    return executeRm(path, flags, true);
+  }
+
+  function executeCopy(
+    path: string,
+    destination: string,
+    flags: string
+  ): string[] {
+    // Make a copy to avoid mutating the original (it's a state variable)
+    const fileSystemCopy = _.cloneDeep(prop.fileSystem);
+    const cwdCopy = fileSystemCopy.getFileSystemObjectFromPath(
+      prop.currentWorkingDirectory.path
+    );
+    const file = path.includes('/')
+      ? prop.fileSystem.getFileSystemObjectFromPath(path)
+      : prop.currentWorkingDirectory.getFileSystemObjectFromPath(path);
+    let destDir;
+    let newFileName;
+
+    // If the path is absolute, find the directory to create the file in
+    // Otherwise, use the current working directory
+    if (destination.includes('/')) {
+      const pathArr = destination.split('/');
+      newFileName = pathArr.pop() || '';
+      if (path.startsWith('/')) {
+        destDir = fileSystemCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      } else {
+        destDir = cwdCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      }
+    } else {
+      newFileName = destination;
+      destDir = cwdCopy;
+    }
+
+    if (!file) {
+      return [`cp: '${path}': No such file or directory`];
+    } else if (file.isDirectory && !flags.toLowerCase().includes('r')) {
+      return [`cp: '${path}' is a directory (not copied)`];
+    } else if (!destDir) {
+      return [`cp: '${destination}': No such file or directory`];
+    } else if (!destDir.isDirectory) {
+      return [`cp: '${destination}': Not a directory`];
+    }
+    const newFile = _.cloneDeep(file);
+    newFile.name = newFileName;
+    destDir.addFileSystemObject(newFile);
+    prop.setFileSystem(fileSystemCopy);
+    prop.setCurrentWorkingDirectory(cwdCopy);
+    return [];
+  }
+
+  function executeFind(path: string, flags: string): [string[], string[]] {
+    if (path || flags) {
+      return [[], ['find: Not implemented']];
+    }
+    return [[], []];
+  }
+
   function isValid(usertyped: string) {
     const userInput = usertyped.trim().split(' ');
     const command = input.length > 1 ? userInput.slice(1, 2).join(' ') : '';
     const args = userInput.slice(2);
-    const flags = args.length > 0 && args[0].startsWith('-') ? args[0] : '';
-    const path =
-      args.length > 0 && args[0].startsWith('-')
-        ? args.slice(1).join(' ')
-        : args.join(' ');
+    let flags = '';
```

Comment:
1. TO-DO - Flags should be an array of strings instead of one string
2. TO-DO - Set up parsing for the flags - please make this in a separate PR with tests (please!)
3. Consideration - might want to refactor out the loop using indexOf() even though performance is very very slightly slower but it'll be a lot more readable (also your loop has mutations and saving state (i))

Reply:
Yeah, the performance drawback of indexOf is really small, so I might add this in the future to improve code readability.